### PR TITLE
wolf-spectrum: init at 1.0.0

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Thanks for @mrVanDalo as this is only copy-paste from here: https://git.ingolf-wagner.de/palo/nixos-config/src/master/pkgs/wolf-spectrum/default.nix

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    mkdir -p $out/lib/vst
+    mkdir -p $out/bin/
+    cp -r bin/wolf-spectrum.lv2    $out/lib/lv2/
+    #cp -r bin/wolf-spectrum-dssi*  $out/lib/dssi/
+    cp -r bin/wolf-spectrum-vst.so $out/lib/vst/
+    cp -r bin/wolf-spectrum        $out/bin/
+  '';
+
+  meta = with pkgs.stdenv.lib; {
+    homepage = "https://pdesaulniers.github.io/wolf-shaper/";
```
Comment (magnetophon) [inline pkgs/applications/audio/wolf-spectrum/default.nix]: You might wanna check the homepage, description and maintainer.

**Review by magnetophon — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/wolf-spectrum/default.nix (hunk 1 above)
Comment (matthiasbeyer) [reply pkgs/applications/audio/wolf-spectrum/default.nix]: Whops, sorry 'bout that.

**Review by magnetophon — APPROVED** [review]

**Comment by magnetophon** [comment]
Works for me!

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  src = pkgs.fetchFromGitHub {
+    owner = "pdesaulniers";
+    repo = "wolf-spectrum";
+    rev = "v${version}";
+    sha256 = "17db1jlj7vb1xyvkdhhrsvdbwb7jqw6i4168cdvlj3yvn2ra8gpm";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = with pkgs; [ pkgconfig ];
```
Comment (jonringer) [inline pkgs/applications/audio/wolf-spectrum/default.nix]: canonical name over alias
```suggestion
  nativeBuildInputs = with pkgs; [ pkg-config ];
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{ pkgs
+, pkgconfig
```
Comment (jonringer) [inline pkgs/applications/audio/wolf-spectrum/default.nix]:
```suggestion
, pkg-config
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{ pkgs
```
Comment (jonringer) [inline pkgs/applications/audio/wolf-spectrum/default.nix]: avoid importing pkgs namespace
```suggestion
{ stdenv
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+, pkgconfig
+, libjack2
+, lv2
+, xorg
+, liblo
+, libGL
+}:
+
+pkgs.stdenv.mkDerivation rec {
```
Comment (jonringer) [inline pkgs/applications/audio/wolf-spectrum/default.nix]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    #mkdir -p $out/lib/dssi
+    mkdir -p $out/lib/vst
+    mkdir -p $out/bin/
+    cp -r bin/wolf-spectrum.lv2    $out/lib/lv2/
+    #cp -r bin/wolf-spectrum-dssi*  $out/lib/dssi/
+    cp -r bin/wolf-spectrum-vst.so $out/lib/vst/
+    cp -r bin/wolf-spectrum        $out/bin/
+  '';
+
+  meta = with pkgs.stdenv.lib; {
```
Comment (jonringer) [inline pkgs/applications/audio/wolf-spectrum/default.nix]:
```suggestion
  meta = with stdenv.lib; {
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2
+    #mkdir -p $out/lib/dssi
+    mkdir -p $out/lib/vst
+    mkdir -p $out/bin/
+    cp -r bin/wolf-spectrum.lv2    $out/lib/lv2/
+    #cp -r bin/wolf-spectrum-dssi*  $out/lib/dssi/
+    cp -r bin/wolf-spectrum-vst.so $out/lib/vst/
+    cp -r bin/wolf-spectrum        $out/bin/
```
Comment (jonringer) [inline pkgs/applications/audio/wolf-spectrum/default.nix]: is there anyway to check that these are the correct install paths? Also, i don't see any headers being installed

Comment (matthiasbeyer) [reply pkgs/applications/audio/wolf-spectrum/default.nix]: Why would you need any headers?

What do you mean by correct install paths? I can tell you that this plugin worked in ardour5 with this setup... not sure what you want. :smile:

Comment (jonringer) [reply pkgs/applications/audio/wolf-spectrum/default.nix]: I guess that's what i was looking for

**Comment by magnetophon** [comment]
still important to me

**Comment by mrVanDalo** [comment]
@matthiasbeyer there is a merge conflict here.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 83579` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wolf-spectrum</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  makeFlags =
+    [ "BUILD_LV2=true" "BUILD_DSSI=false" "BUILD_VST2=true" "BUILD_JACK=true" ];
+
+  patchPhase = ''
+    patchShebangs ./dpf/utils/generate-ttl.sh
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2
+    #mkdir -p $out/lib/dssi
```
Comment (nh2) [inline pkgs/applications/audio/wolf-spectrum/default.nix:35]: Randomly commented-out stuff, and this `installPhase` in general: Please add some comments that describe what's (not) copied and why.

I suspect from `"BUILD_DSSI=false"` above that it doesn't build and you might want it to build in the future -- a comment on `"BUILD_DSSI=false"` might be excellent for that (you may also want to make the `makeFlags` above multi-line to best be able to place your comment).

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    rev = "v${version}";
+    sha256 = "17db1jlj7vb1xyvkdhhrsvdbwb7jqw6i4168cdvlj3yvn2ra8gpm";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ libjack2 lv2 xorg.libX11 liblo libGL xorg.libXcursor ];
+
+  makeFlags =
+    [ "BUILD_LV2=true" "BUILD_DSSI=false" "BUILD_VST2=true" "BUILD_JACK=true" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/wolf-spectrum/default.nix:27]:
```suggestion
  makeFlags = [
    "BUILD_LV2=true"
    "BUILD_DSSI=false"
    "BUILD_VST2=true"
    "BUILD_JACK=true"
  ];
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    #mkdir -p $out/lib/dssi
+    mkdir -p $out/lib/vst
+    mkdir -p $out/bin/
+    cp -r bin/wolf-spectrum.lv2    $out/lib/lv2/
+    #cp -r bin/wolf-spectrum-dssi*  $out/lib/dssi/
+    cp -r bin/wolf-spectrum-vst.so $out/lib/vst/
+    cp -r bin/wolf-spectrum        $out/bin/
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/wolf-spectrum/default.nix:44]:
```suggestion
  meta = with lib; {
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    homepage = "https://github.com/pdesaulniers/wolf-spectrum";
+    description = "Real-time 2D spectrogram plugin (LV2, VST and Jack)";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/wolf-spectrum/default.nix:47]: gpl3Plus or gpl3Only?

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 12]
```diff
@@ -0,0 +1,51 @@
+{ stdenv
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/wolf-spectrum/default.nix:1]:
```suggestion
{ stdenv
, lib
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+    patchShebangs ./dpf/utils/generate-ttl.sh
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2
+    #mkdir -p $out/lib/dssi
+    mkdir -p $out/lib/vst
+    mkdir -p $out/bin/
+    cp -r bin/wolf-spectrum.lv2    $out/lib/lv2/
+    #cp -r bin/wolf-spectrum-dssi*  $out/lib/dssi/
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/wolf-spectrum/default.nix:44]:
```suggestion
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+  patchPhase = ''
+    patchShebangs ./dpf/utils/generate-ttl.sh
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2
+    #mkdir -p $out/lib/dssi
+    mkdir -p $out/lib/vst
+    mkdir -p $out/bin/
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/wolf-spectrum/default.nix:42]:
```suggestion
    mkdir -p $out/lib/lv2 $out/lib/vst $out/bin/
```

File: pkgs/applications/audio/wolf-spectrum/default.nix [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+    cp -r bin/wolf-spectrum-vst.so $out/lib/vst/
+    cp -r bin/wolf-spectrum        $out/bin/
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/pdesaulniers/wolf-spectrum";
+    description = "Real-time 2D spectrogram plugin (LV2, VST and Jack)";
+    license = licenses.gpl3Plus;
+    maintainers = [ maintainers.matthiasbeyer ];
+    platforms = [ "i686-linux" "x86_64-linux" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/wolf-spectrum/default.nix:54]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/applications/audio/wolf-spectrum/default.nix (hunk 13 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/wolf-spectrum/default.nix:44]: You forgot to remove the comment.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**